File: solutions/clojure/grade-school/1/src/grade_school.clj
(ns grade-school)

(defn merge-class-vecs [v1 v2]
  (into [] (concat v1 v2))
  )

(defn grade [school grade]  
 (if (contains? school grade) 
   (school grade)
   []) 
)

(defn add [school name grade]  
    (merge-with merge-class-vecs  school {grade [name]})
)

(defn sort-students [school]
  (into {} (for [[k v] school]
             [k (into [] (sort v))]))
  )

(defn sorted [school]  
  (into (sorted-map) (sort-students school) )
)

File: solutions/python/grade-school/1/grade_school.py
class School:
    def __init__(self):
        self.school={}
        self.aditions=[]

    def add_student(self, name, grade):
        if name not in self.roster():
            self.school[grade] = self.school.get(grade, []) + [name]  
            self.aditions.append(True)
        else:
            self.aditions.append(False)
            
    def roster(self):
        return [name 
                for grade in sorted(self.school.keys())
                for name in sorted(self.school[grade])]

    def grade(self, grade_number):
        return sorted(self.school.get(grade_number, []))
        

    def added(self):
        return self.aditions
